redmond-dining-bot/Controllers/MessagesController.cs

```diff
@@ -29,38 +36,97 @@ public class MessagesController : ApiController
+                                ResetShuttleVariables();
+                            }
+                            break;
+
+                      case "no":
+                            if (ContextCallShuttle)
+                            {
+                                BotResponse = "I'm sorry, let's start over then. What do you want me to do?";
+                                ResetShuttleVariables();
+                            }
+                            break;
+
+                      case "schedule shuttle":
+                            BotResponse = "I need to know where to pick you up and drop you off. Please state from where to where do you need the shuttle";
+                            break;
+
+                     default:
+                            BotResponse = "Sorry, I can't understand you...";
+                            break;
+                        }
+                    
+                }
+
+                else if (diLUIS.intents.Count() > 0 && diLUIS.entities.Count() > 0)
                 {
+                    //Handle intents with entities
                     switch (diLUIS.intents[0].intent)
                     {
                         case "list-all-cafe": //find-food is an intent from LUIS
-                            diningoption = await GetAllCafes();
+                            BotResponse = await GetAllCafes();
                             break;
 
                         case "find-food": //find-food is an intent from LUIS
-                            diningoption = await GetCafeForItem(diLUIS.entities[0].entity);
+                            BotResponse = await GetCafeForItem(diLUIS.entities[0].entity);
                             break;
 
                         // change this back to GetMenu if test does not work out
                         case "find-menu": //find-food is an intent from LUIS
-                            diningoption = await GetCafeMenu(diLUIS.entities[0].entity);
+                            BotResponse = await GetCafeMenu(diLUIS.entities[0].entity);
                             break;
 
+                        case "schedule shuttle":
+                            if (diLUIS.entities[0].type == "Destination Building" && diLUIS.entities[1].type == "Origin Building")
+                            {
+                                BotResponse = await SetShuttleRequest(diLUIS.entities[0].entity, diLUIS.entities[1].entity);
+                            }
+                            else
+                            {
+                                //bot ask user to clearly state from where do you want me to take you and to where. 
+                                if (diLUIS.entities[0].type == "Destination Building")
+                                {
+                                    //if destination given
+                                    BotResponse = "I need to know where to pick you up. Can you state from where to where do you need the shuttle?";
```

Comment:
Bot doesn't support asking the user for a missing field (origin or destination) at this point. It rejects the command and asks the user to try again.